controller/contour/daemonset.go

```diff
@@ -202,28 +189,44 @@ func DesiredDaemonSet(contour *operatorv1alpha1.Contour, contourImage, envoyImag
 					},
 				},
 			},
+			ReadinessProbe: &corev1.Probe{
+				FailureThreshold: int32(3),
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Scheme: corev1.URISchemeHTTP,
+						Path:   "/ready",
+						Port:   intstr.IntOrString{IntVal: int32(8002)},
+					},
+				},
+				InitialDelaySeconds: int32(3),
+				PeriodSeconds:       int32(4),
+				SuccessThreshold:    int32(1),
+				TimeoutSeconds:      int32(1),
+			},
 			Ports: []corev1.ContainerPort{
 				{
 					Name:          "http",
-					ContainerPort: 80,
+					ContainerPort: int32(httpPort),
+					HostPort:      int32(httpPort),
 					Protocol:      "TCP",
 				},
 				{
 					Name:          "https",
-					ContainerPort: 443,
+					ContainerPort: int32(httpsPort),
+					HostPort:      int32(httpsPort),
 					Protocol:      "TCP",
 				},
 			},
 			VolumeMounts: []corev1.VolumeMount{
 				{
 					Name:      envoyCertsVolName,
 					MountPath: filepath.Join("/", envoyCertsVolMntDir),
-					ReadOnly:  true,
+					ReadOnly:  false,
```

Comment:
Does the certs volume really need to be writable?

Reply:
Thanks for catching this. No, the cert vol should be read only.